src/tasks/tasks.py

```diff
@@ -2224,3 +2224,26 @@ def get_metrics(self, reset=False):
         f1 = (f11 + f12 + f13) / 3
         return {'accuracy': acc, 'f1': f1, 'precision': pcs, 'recall': rcl}
 
+@register_task('winograd-coreference', rel_path = 'winograd-coref')
+class WinogradCoreferenceTask(SpanClassificationTask):
+    def __init__(self, path,  **kw):
+        self._files_by_split = {'train': "train.json", 'val': "val.json",'test': "test_final.json"}
+        self.num_spans = 2
+        super().__init__(files_by_split=self._files_by_split, label_file="labels.txt", path=path, **kw)
+        self.val_metric = "%s_acc" % self.name 
+
+    def update_metrics(self, logits, labels, tagmask=None):
+        logits, labels = logits.detach(), labels.detach()
+        def one_hot_v(batch, depth=2):
```

Comment:
Comment: What does this do?

Reply:
This creates one hot embeddings given the logits , which is what the F1 scorer and acc scorer expects.

Reply:
Just suggesting you comment in the code.